core/templates/pages/landing-pages/topic-landing-page/topic-landing-page.component.ts

```diff
@@ -17,118 +17,146 @@
-          imageWebpFilename: 'suitable_for_all.webp',
-          imageAlt: 'Lesson viewers and learners.'
-        }];
-      };
-      var topicData = null;
-
-      ctrl.getLessonQualityImageSrc = function(filename) {
-        return UrlInterpolationService.getStaticImageUrl(
-          UrlInterpolationService.interpolateUrl(
-            '/landing/<filename>', {filename: filename}));
-      };
-
-      ctrl.onClickGetStartedButton = function() {
-        var collectionId = topicData.collectionId;
-        SiteAnalyticsService.registerOpenCollectionFromLandingPageEvent(
-          collectionId);
-        $timeout(function() {
-          WindowRef.nativeWindow.location = UrlInterpolationService
-            .interpolateUrl('/collection/<collectionId>', {
-              collectionId: collectionId
-            });
-        }, 150);
-      };
-
-      ctrl.onClickLearnMoreButton = function() {
-        $timeout(function() {
-          WindowRef.nativeWindow.location = '/community-library';
-        }, 150);
-      };
-
-      ctrl.$onInit = function() {
-        var pathArray = WindowRef.nativeWindow.location.pathname.split('/');
-        var subjectName = pathArray[1];
-        var topicName = pathArray[2];
-        topicData = TOPIC_LANDING_PAGE_DATA[subjectName][topicName];
-        ctrl.topicTitle = topicData.topicTitle;
-
-        ctrl.lessonsQualities = getLessonQualities();
-        ctrl.backgroundBannerUrl = (
-          UrlInterpolationService.getStaticImageUrl(
-            '/background/bannerB.svg'));
-
-        var topicImageUrlTemplate = '/landing/<subject>/<topic>/<filename>';
-        ctrl.lessonInDevicesPngImageSrc = (
-          UrlInterpolationService.getStaticImageUrl(
-            UrlInterpolationService.interpolateUrl(
-              topicImageUrlTemplate, {
-                subject: subjectName,
-                topic: topicName,
-                filename: 'lesson_in_devices.png'
-              })));
-        ctrl.lessonInDevicesWebpImageSrc = (
-          UrlInterpolationService.getStaticImageUrl(
-            UrlInterpolationService.interpolateUrl(
-              topicImageUrlTemplate, {
-                subject: subjectName,
-                topic: topicName,
-                filename: 'lesson_in_devices.webp'
-              })));
-
-        PageTitleService.setPageTitle(
-          [ctrl.topicTitle, topicData.topicTagline, 'Oppia'].join(' | '));
-      };
-    }
-  ]
-});
+
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+import { PageTitleService } from 'services/page-title.service';
+import { TopicLandingPageConstants } from
+  'pages/landing-pages/topic-landing-page/topic-landing-page.constants';
```

Comment:
'p' < 's', this implies 'pages' < 'services'

Reply:
Done, separated the constant file from the other services too